sql-plugin/src/main/spark341db/scala/com/nvidia/spark/rapids/shims/GpuBatchScanExec.scala

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*** spark-rapids-shim-json-lines
+{"spark": "341db"}
+spark-rapids-shim-json-lines ***/
+package com.nvidia.spark.rapids.shims
+
+import com.google.common.base.Objects
+import com.nvidia.spark.rapids.{GpuBatchScanExecMetrics, GpuScan}
+
+import org.apache.spark.SparkException
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{AttributeReference, DynamicPruningExpression, Expression, Literal, SortOrder}
+import org.apache.spark.sql.catalyst.plans.QueryPlan
+import org.apache.spark.sql.catalyst.plans.physical.{KeyGroupedPartitioning, SinglePartition}
+import org.apache.spark.sql.catalyst.util.{truncatedString, InternalRowComparableWrapper}
+import org.apache.spark.sql.connector.catalog.Table
+import org.apache.spark.sql.connector.read._
+import org.apache.spark.sql.execution.datasources.rapids.DataSourceStrategyUtils
+import org.apache.spark.sql.execution.datasources.v2._
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+case class GpuBatchScanExec(
+    output: Seq[AttributeReference],
+    @transient scan: GpuScan,
+    runtimeFilters: Seq[Expression],
+    keyGroupedPartitioning: Option[Seq[Expression]],
+    ordering: Option[Seq[SortOrder]], 
+    @transient table: Table,
+    reusesFileListingResultsSourceNode: Option[BatchScanExec])
+    extends DataSourceV2ScanExecBase with GpuBatchScanExecMetrics {
```

Comment:
There seem to be some missing fields here compared to what I'm seeing in Spark 3.4 source.  The latter has `commonPartitionValues`, `applyPartialClustering`, and `replicatePartitions` along with logic referencing these in `inputRDD`, `equals`, and elsewhere.